es-core/src/Settings.cpp

```diff
@@ -115,7 +115,7 @@ void Settings::setDefaults()
 	mStringMap["SlideshowScreenSaverBackgroundAudioFile"] = Utils::FileSystem::getHomePath() + "/.emulationstation/slideshow/audio/slideshow_bg.wav";
 	mBoolMap["SlideshowScreenSaverCustomImageSource"] = false;
 	mStringMap["SlideshowScreenSaverImageDir"] = Utils::FileSystem::getHomePath() + "/.emulationstation/slideshow/image";
-	mStringMap["SlideshowScreenSaverImageFilter"] = ".png,.jpg";
+	mStringMap["SlideshowScreenSaverImageFilter"] = ".png,.jpg,.mp4,.avi";
```

Comment:
Hm. How does this relate to the other "videoExtensions" variable - especially as this string can be changed by the user in the settings file at least, as well as the options menu I believe?
Also, users will probably already have this setting saved as ".png,.jpg", and this change will only be effective for new users.

Reply:
This does not directly relate to the "videoExtensions" variable, the strings in that vector are hardcoded. I only included ".mp4" and ".avi" as those are two common video formats, just as ".png" and ".jpg" are two common image formats. In theory any format that the video player supports should work if it is added both here and to "videoExtensions".

Yes, this would only impact new users. Continuing users would need to change the settings in the menu themselves to prevent the files from being blocked by the filter, unless someone knows of another way. Since this feature is supposed to add support for ".mp4" and ".avi" files in the slideshow mode, I thought it made sense to add them as part of the default settings. Otherwise even new users would need to change their settings to enable the feature.

Reply:
Correct - what I was challenging was for us to come up with a way to make it work for both, so that nobody has to change these. I imagine the majority of users will be "continuing users".

Reply:
Ok. I think one way or another the new file extensions have to get written into the settings file, right? You could override the file filter in SystemScreenSaver but then the settings would misrepresent what's actually happening.

I was looking at processBackwardCompatability() in Settings, but now I'm not so sure
https://github.com/RetroPie/EmulationStation/blob/c612f23b03c8583dce479039afcc83a2cc242049/es-core/src/Settings.cpp#L249
You could check for and append the new extensions after load, but that would make them impossible to remove and defeat the point of the setting.
That makes me think it would have to be a one time change during the update, so the user could remove it if needed. And if that is the case, I'll need to do a bit more research. Thoughts?

Reply:
I suppose that's a fair assessment. Some thoughts:
- Name this "SlideshowScreenSaverMediaFilter" (I think keeping calling it "Image" is misleading now, and it should probably be reflected in the menu at least).
- If it doesn't exist, initialize it from concatenating the values in SlideshowScreenSaverImageFilter (loaded or default, if none) and concatenate mp4, avi . After it gets saved, it should work moving forward.

Maybe this is an extremely stupid or convoluted idea, but that's an angle I'd imagine could work. That processing could either be done in `processBackwardCompatibility()` or on load, I don't quite recall that code.

Don't take these as definitive suggestions, more as options to consider.

Thanks!

Reply:
I think that makes a lot of sense, if renaming the setting needs to happen then it will need to be processed in `processBackwardCompatibility()` anyway, so might as well do some work on it then. It's just taking me a minute to get done now that work has ramped up again. Thank you!

Reply:
Hey @pjft, would you mind taking a look at [this branch](https://github.com/TheMasterWho/EmulationStation/tree/renameImageToMedia) as a proposed solution to this issue? I might have gone a little overboard, so I didn't want to directly commit those changes into this merge request. I tried my best to reduce friction for new and continuing users, video extensions are now their own setting rather than hardcoding them in and expecting a user to modify source code if they want to add support for a different file type. I also tried to rename any instances of the Slideshow ScreenSaver to use 'media' rather than 'image' to minimize confusion. Thanks!

(Also please ignore the commit dates on that branch, those were made today. My VM is just way messed up)

Reply:
Done, squashed into 300828e. Thank you!